### ISSUE TITLE
mt5-base-ViT-B-32: Couldn't instantiate tokenizer

### ISSUE THREAD

**Issue by lopho** [mlfoundations/open_clip#246]
This also fails the training & inference test for 'mt5-base-ViT-B-32'

```py
open_clip.get_tokenizer('mt5-base-ViT-B-32')
```
without sentencepiece
```
  File "/REDACTED/open_clip/src/open_clip/factory.py", line 75, in get_tokenizer
    tokenizer = HFTokenizer(config['text_cfg']['hf_tokenizer_name']) if 'hf_tokenizer_name' in config['text_cfg'] else tokenize
  File "/REDACTED/open_clip/src/open_clip/tokenizer.py", line 192, in __init__
    self.tokenizer = AutoTokenizer.from_pretrained(tokenizer_name)
  File "/REDACTED/open_clip/.venv/lib/python3.9/site-packages/transformers/models/auto/tokenization_auto.py", line 619, in from_pretrained
    return tokenizer_class.from_pretrained(pretrained_model_name_or_path, *inputs, **kwargs)
  File "/REDACTED/open_clip/.venv/lib/python3.9/site-packages/transformers/tokenization_utils_base.py", line 1777, in from_pretrained
    return cls._from_pretrained(
  File "/REDACTED/open_clip/.venv/lib/python3.9/site-packages/transformers/tokenization_utils_base.py", line 1932, in _from_pretrained
    tokenizer = cls(*init_inputs, **init_kwargs)
  File "/REDACTED/open_clip/.venv/lib/python3.9/site-packages/transformers/models/t5/tokenization_t5_fast.py", line 134, in __init__
    super().__init__(
  File "/REDACTED/open_clip/.venv/lib/python3.9/site-packages/transformers/tokenization_utils_fast.py", line 120, in __init__
    raise ValueError(
ValueError: Couldn't instantiate the backend tokenizer from one of: 
(1) a `tokenizers` library serialization file, 
(2) a slow tokenizer instance to convert or 
(3) an equivalent slow tokenizer class to instantiate and convert. 
You need to have sentencepiece installed to convert a slow tokenizer to a fast one.
```
with sentencepiece & latest protobuf (4.21.9)
```
  File "/REDACTED/open_clip/src/open_clip/factory.py", line 75, in get_tokenizer
    tokenizer = HFTokenizer(config['text_cfg']['hf_tokenizer_name']) if 'hf_tokenizer_name' in config['text_cfg'] else tokenize
  File "/REDACTED/open_clip/src/open_clip/tokenizer.py", line 192, in __init__
    self.tokenizer = AutoTokenizer.from_pretrained(tokenizer_name)
  File "/REDACTED/open_clip/.venv/lib/python3.9/site-packages/transformers/models/auto/tokenization_auto.py", line 619, in from_pretrained
    return tokenizer_class.from_pretrained(pretrained_model_name_or_path, *inputs, **kwargs)
  File "/REDACTED/open_clip/.venv/lib/python3.9/site-packages/transformers/tokenization_utils_base.py", line 1777, in from_pretrained
    return cls._from_pretrained(
  File "/REDACTED/open_clip/.venv/lib/python3.9/site-packages/transformers/tokenization_utils_base.py", line 1932, in _from_pretrained
    tokenizer = cls(*init_inputs, **init_kwargs)
  File "/REDACTED/open_clip/.venv/lib/python3.9/site-packages/transformers/models/t5/tokenization_t5_fast.py", line 134, in __init__
    super().__init__(
  File "/REDACTED/open_clip/.venv/lib/python3.9/site-packages/transformers/tokenization_utils_fast.py", line 114, in __init__
    fast_tokenizer = convert_slow_tokenizer(slow_tokenizer)
  File "/REDACTED/open_clip/.venv/lib/python3.9/site-packages/transformers/convert_slow_tokenizer.py", line 1162, in convert_slow_tokenizer
    return converter_class(transformer_tokenizer).converted()
  File "/REDACTED/open_clip/.venv/lib/python3.9/site-packages/transformers/convert_slow_tokenizer.py", line 438, in __init__
    from .utils import sentencepiece_model_pb2 as model_pb2
  File "/REDACTED/open_clip/.venv/lib/python3.9/site-packages/transformers/utils/sentencepiece_model_pb2.py", line 92, in <module>
    _descriptor.EnumValueDescriptor(
  File "/REDACTED/open_clip/.venv/lib/python3.9/site-packages/google/protobuf/descriptor.py", line 755, in __new__
    _message.Message._CheckCalledFromGeneratedFile()
TypeError: Descriptors cannot not be created directly.
If this call came from a _pb2.py file, your generated code is out of date and must be regenerated with protoc >= 3.19.0.
If you cannot immediately regenerate your protos, some other possible workarounds are:
 1. Downgrade the protobuf package to 3.20.x or lower.
 2. Set PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION=python (but this will use pure-Python parsing and will be much slower).

More information: https://developers.google.com/protocol-buffers/docs/news/2022-05-06#python-updates
```

It works with sentencepiece and protobuf==3.20.0.

Adding this information to the README or as a custom error message should be enough. Otherwise, users would have to run this three times until all the necessary information has been gathered from the exceptions.

**Comment by rom1504** [comment]
Yes the fix is protobuf==3.20.0
Just run pip install -e .
It will install the right versions of dependencies

**Comment by lopho** [comment]
Yes, was a quickshot, sorry.
Also latest protobuf that can be supported is  `3.20.3`
